### PR TITLE
AutoGen logging using Azure Cosmos DB

### PR DESCRIPTION
**Pull request by wmwxwa**
## Why are these changes needed?

Adding the connector to use Cosmos DB for logging

## Related issue number

N/A

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — CHANGES_REQUESTED** [review]
Thanks for the PR! A couple of things to add:

1. Add handling for the cosmosdb logger in the factory method in https://github.com/microsoft/autogen/blob/5a96dc2c2969a98f08d7fc4853b3419f61065609/autogen/logger/logger_factory.py#L11-L12 with a try-except block to catch import error when cosmos db package is not installed.
2. In setup.py add a new extra_require entry for cosmosdb similar to redis: https://github.com/microsoft/autogen/blob/5a96dc2c2969a98f08d7fc4853b3419f61065609/setup.py#L49-L50
3. Add a unit test for the logger. @tonybaloney do you want to comment on how to add a mock test for cosmos db usage?

File: autogen/logger/cosmos_db_logger
Comment (jackgerrits) [inline autogen/logger/cosmos_db_logger:1]: missing `.py` file extension

File: autogen/logger/cosmos_db_logger [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+from autogen.logger.logger_utils import get_current_ts, to_dict
+
+if TYPE_CHECKING:
+    from autogen.agent.conversable_agent import ConversableAgent
+    from autogen.wrapper.openai_wrapper import OpenAIWrapper
+
+logger = logging.getLogger(__name__)
+
+class CosmosDBLogger(BaseLogger):
+    def __init__(self, config: Dict[str, Any]):
```
Comment (jackgerrits) [inline autogen/logger/cosmos_db_logger:15]: Please make `config` a typed dictionary

**Review by jackgerrits — CHANGES_REQUESTED** [review]

File: autogen/logger/cosmos_db_logger.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import uuid
+import logging
+from typing import Any, Dict, Union, TYPE_CHECKING, TypedDict
+from azure.cosmos import CosmosClient
```
Comment (cheng-tan) [inline autogen/logger/cosmos_db_logger.py:4]: dependency needs to be added in setup.py as extras_require like redis

**Review by cheng-tan — COMMENTED** [review]
Thank you for the PR.
To fix the pre-commit failure: run `pre-commit run --all-files`, 
to fix the type issues run `mypy autogen/logger/cosmos_db_logger.py`.

Can you also add some unit test for cosmos db logger like Eric suggested above? Existing logging unit test is under test/test_logging.py

File: autogen/logger/logger_factory.py [hunk 3]
```diff
@@ -3,8 +3,22 @@
...
+try:
+    from autogen.logger.cosmos_db_logger import CosmosDBConfig, CosmosDBLogger
+
+    cosmos_imported = True
+except ImportError:
+    cosmos_imported = False
+
 __all__ = ("LoggerFactory",)
 
+try:
```
Comment (ekzhu) [inline autogen/logger/logger_factory.py:15]: Duplicate code blocks?

**Comment by wmwxwa** [comment]
[ ] Add content on Azure Cosmos DB to https://github.com/microsoft/autogen/blob/main/notebook/agentchat_logging.ipynb

**Comment by wmwxwa** [comment]
Cannot revolve conflict. Continuing the work under [PR 2760](https://github.com/microsoft/autogen/pull/2760)

**Pull request closed without merging**